modin/core/io/text/csv_glob_dispatcher.py

```diff
@@ -383,35 +384,39 @@ def get_path(cls, file_path: str) -> list:
         list
             List of strings of absolute file paths.
         """
-        if fsspec.core.split_protocol(file_path)[0] in ("s3", "S3"):
-            # `file_path` may start with a capital letter, which isn't supported by `fsspec.core.url_to_fs` used below.
-            file_path = file_path[0].lower() + file_path[1:]
-
-            from botocore.exceptions import (
-                NoCredentialsError,
-                EndpointConnectionError,
-                ConnectTimeoutError,
-            )
-
-            def get_file_path(fs_handle) -> List[str]:
-                file_paths = fs_handle.glob(file_path)
-                if len(file_paths) == 0 and not fs_handle.exists(file_path):
-                    raise FileNotFoundError(f"Path <{file_path}> isn't available.")
-                s3_addresses = [f"s3://{path}" for path in file_paths]
-                return s3_addresses
-
-            fs, _ = fsspec.core.url_to_fs(file_path)
-            try:
-                return get_file_path(fs)
-            except (NoCredentialsError, EndpointConnectionError, ConnectTimeoutError):
-                pass
-            fs, _ = fsspec.core.url_to_fs(file_path, anon=True)
-            return get_file_path(fs)
-        else:
+        if fsspec.core.split_protocol(file_path)[0] not in _SUPPORTED_PROTOCOLS:
             relative_paths = glob.glob(file_path)
             abs_paths = [os.path.abspath(path) for path in relative_paths]
             return abs_paths
 
+        # `file_path` may start with a capital letter, which isn't supported by `fsspec.core.url_to_fs` used below.
+        file_path = file_path[0].lower() + file_path[1:]
+
+        from botocore.exceptions import (
+            NoCredentialsError,
+            EndpointConnectionError,
+            ConnectTimeoutError,
+        )
+
+        def get_file_path(fs_handle) -> List[str]:
+            file_paths = fs_handle.glob(file_path)
+            if len(file_paths) == 0 and not fs_handle.exists(file_path):
+                raise FileNotFoundError(f"Path <{file_path}> isn't available.")
+            s3_addresses = [f"s3://{path}" for path in file_paths]
```

Comment:
This feels wrong, what if original protocol wasn't `S3`-like? Previous version was explicitly checking for S3 only...

Reply:
This version also explicitly checking on supported protocols in the beginning of `get_path`. In case protocol is n't {"s3", "S3"} this line won't be run.

Reply:
Could we add a TODO here to change this to be more generalizable in the future? Or it might be good to at least mention this in the issue I created when we try to add more support to things other than s3.

Reply:
oh, I've missed that this file has different `_SUPPORTED_PROTOCOLS`, sorry! 🙃

Reply:
@pyrito I mentioned this thread in #4742 instead of TODO because the PR was already merged.

Reply:
@prutskov all good. Thanks!